src/Raven.Server/Documents/Indexes/Static/JavaScript/AttachmentObjectInstance.cs

```diff
@@ -83,9 +88,9 @@ public override PropertyDescriptor GetOwnProperty(JsValue property)
                 else if (property == nameof(IAttachmentObject.Size))
                     value = new PropertyDescriptor(_attachment.Size, writable: false, enumerable: false, configurable: false);
                 else if (property == nameof(IAttachmentObject.Flags))
-                    value = new PropertyDescriptor((int)_attachment.Flags, writable: false, enumerable: false, configurable: false);
+                    value = new PropertyDescriptor(new JsString(_attachment.Flags.ToString()), writable: false, enumerable: false, configurable: false);
                 else if (property == nameof(IAttachmentObject.RetireAt))
-                    value = _attachment.RetireAt.HasValue ? new PropertyDescriptor(new JsDate(_engine, _attachment.RetireAt.Value), writable: false, enumerable: false, configurable: false) : null;
+                    value = _attachment.RetireAt is DateTime dt ? new PropertyDescriptor(new JsDate(_engine, dt), writable: false, enumerable: false, configurable: false) : null;
```

Comment:
why the casting? cannot we keep `_attachment.RetireAt.HasValue` and use `t.RetireAt.Value` ?

Reply:
never mind this is fine